### PR TITLE
kojiapi: move to echo

### PR DESCRIPTION
**Pull request by teg**
Unify with the worker API and move from the chi to the echo libraries.

### CODE REVIEW

File: internal/kojiapi/api/api.go [hunk 1]
```diff
@@ -0,0 +1,3 @@
+//go:generate go run github.com/deepmap/oapi-codegen/cmd/oapi-codegen -package=api -generate types,server -o api.gen.go openapi.yml
```
Comment (Gundersanne) [inline internal/kojiapi/api/api.go:1]: Hm `client` was remote from -generate here, do we not use it?

File: internal/kojiapi/server.go [hunk 2]
```diff
@@ -24,66 +23,70 @@ import (
...
-		return
+		return echo.NewHTTPError(http.StatusBadRequest, "Invalid Koji server: %s", request.Koji.Server)
 	}
-	creds, exists := server.kojiServers[kojiServer.Hostname()]
+	creds, exists := h.server.kojiServers[kojiServer.Hostname()]
 	if !exists {
-		http.Error(w, fmt.Sprintf("Koji server has not been configured: %s", kojiServer.Hostname()), http.StatusBadRequest)
-		return
-
+		echo.NewHTTPError(http.StatusBadRequest, "Koji server has not been configured: %s", kojiServer.Hostname())
```
Comment (Gundersanne) [inline internal/kojiapi/server.go:89]: missing return statement?

File: internal/kojiapi/api/api.go (hunk 1 above)
Comment (larskarlitski) [reply internal/kojiapi/api/api.go:1]: If we don't use it, I recommend not generating it. The client code generated by `deepmap` is all over the place. I've opted doing it manually in the `worker` package as well.

Comment (teg) [reply internal/kojiapi/api/api.go:1]: The code that used the client api was dropped (it was an unused test). Generating it still was a left-over.

File: internal/kojiapi/server.go (hunk 2 above)
Comment (teg) [reply internal/kojiapi/server.go]: Fixed.

**Comment by teg** [comment]
Test failures are due to #997, and unrelated to this PR.

File: internal/kojiapi/server.go [hunk 3]
```diff
@@ -23,66 +23,70 @@ import (
...
-		return
+		return err
 	}
 
-	d := server.distros.GetDistro(request.Distribution)
+	d := h.server.distros.GetDistro(request.Distribution)
 	if d == nil {
-		http.Error(w, fmt.Sprintf("Unsupported distribution: %s", request.Distribution), http.StatusBadRequest)
-		return
+		return echo.NewHTTPError(http.StatusBadRequest, "Unsupported distribution: %s", request.Distribution)
```
Comment (larskarlitski) [inline internal/kojiapi/server.go:80]: Unfortunately, this function does not take format strings (cf [echo.NewHTTPError](https://pkg.go.dev/github.com/labstack/echo/v4#NewHTTPError). I realized this too late myself when working on the worker API and fixed it in https://github.com/osbuild/osbuild-composer/commit/ccebfb014c4bb53870d1c876ab4868c936d5178a.

**Review by larskarlitski — CHANGES_REQUESTED** [review]
Thanks! The failure is unrelated indeed (sad that we're having to say this more often again...)

Super sorry: I overlooked something in the first round. Comment inline.

**Review by larskarlitski — APPROVED** [review]
Force-pushed a fix for my latest nitpick myself. Was an easy one.

@teg, please have a look at these changes before merging. Otherwise it's all good. Thanks again!

**Pull request merged**